sdk/overview.md

```diff
@@ -0,0 +1,124 @@
+# Abstract
+
+This document outlines a proposal for mobile service SDK 
+
+# Goals
+
+Main goal for this document is to provide full overview of the future mobile SDK.
+Document outlines a proposal for mobile service SDK's and all related elements.
+Proposal basing on industry standards for building multi service SDK's. 
+
+## Primary Goals
+
+Provide top level architecture and delivery model for mobile SDK's.
+
+## Non Goals
+
+Describe individual service SDK's:
+This will be done in separate proposals.
+
+# Problem Description
+
+Selecting best architecture used to build mobile SDK . 
+
+## Elements to include
+
+1. Git hosting
+
+How to host SDK's in git.
+Single repository for platform or service etc.?
+
+2. SDK distribution to end users
+Platform specific way of distributing packages
+Can be maven, cocoapods etc. 
+
+3. Documentation for SDK
+Api documentation and top level SDK documentation. 
+
+4. Architecture
+How to allow users to interact with SDK.
+Allow flexibility on network layer etc.
+How to prevent users from including large binary when they need single service.
+ 
+5. Adding common services and addons for all services
+Functionalities like network handling, logging, monitoring and audit, error handling, analytics.
+
+6. Developer tooling 
+How to build or provide (plugins, extensions) to service
+
+7. Size constraints 
+For example method limit on android
+
+8. Platform language
+Supporting multiple flavours (languages per platform) - Swift vs ObjectiveC. Java vs Kotlin etc.
+
+# Proposed Solution
+
+Modular SDK For SDK that can contain multiple non related services 
```

Comment:
I don't really understand this sentence can you elaborate

Reply:
Yes. It's not clear.Going to correct that.

This refers to the services available on the backend - like security, sync etc.
Those can be used separately or in the bundle.